packages/core/src/external/aws/s3.ts

```diff
@@ -217,30 +217,69 @@ export class S3Utils {
 
     return newKey;
   }
-  async uploadFile(
-    bucket: string,
-    key: string,
-    file: Buffer
-  ): Promise<AWS.S3.ManagedUpload.SendData> {
+  async uploadFile({
+    bucket,
+    key,
+    file,
+    contentType,
+  }: {
+    bucket: string;
+    key: string;
+    file: Buffer;
+    contentType?: string;
+  }): Promise<AWS.S3.ManagedUpload.SendData> {
     return new Promise((resolve, reject) => {
-      this._s3.upload(
-        {
-          Bucket: bucket,
-          Key: key,
-          Body: file,
-        },
-        (err, data) => {
-          if (err) {
-            console.error("Error during upload:", err);
-            reject(err);
-          } else {
-            console.log("Upload successful");
-            resolve(data);
-          }
+      const uploadParams: AWS.S3.PutObjectRequest = {
+        Bucket: bucket,
+        Key: key,
+        Body: file,
+      };
+
+      if (contentType) {
+        uploadParams.ContentType = contentType;
```

Comment:
explicitly setting the content type on uploads. This is a bug we had in gateway v1 that caused the whole cx with outdated content types on s3